### PR TITLE
fix(rollup-relayer): avoid compressing "small tip" block

### PR DESCRIPTION
**Pull request by colinlyguo**
### Purpose or design rationale of this PR

Applying related fix: https://github.com/scroll-tech/da-codec/pull/17

- [x] Updating the dependency in `go.mod` after merging the above pr.

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] fix: A bug fix

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change

### CODE REVIEW

**Review by Thegaram — COMMENTED** [review]
Need to consider the error from `ConstructBlobPayload` in `batch-proposer`.

File: rollup/internal/controller/watcher/chunk_proposer_test.go [hunk 1]
```diff
@@ -568,51 +568,53 @@ func testChunkProposerCodecv2Limits(t *testing.T) {
...
+			chainConfig = &params.ChainConfig{}
+		} else if codecVersion == encoding.CodecV1 {
 			chainConfig = &params.ChainConfig{BernoulliBlock: big.NewInt(0)}
+		} else {
+			chainConfig = &params.ChainConfig{BernoulliBlock: big.NewInt(0), CurieBlock: big.NewInt(0)}
 		}
 
 		cp := NewChunkProposer(context.Background(), &config.ChunkProposerConfig{
-			MaxBlockNumPerChunk:             math.MaxUint64,
+			MaxBlockNumPerChunk:             255,
```
Comment (colinlyguo) [inline rollup/internal/controller/watcher/chunk_proposer_test.go:592]: da-codec has added a max 255 blocks per chunk check, thus fixing this unit test.

File: rollup/internal/utils/utils.go [hunk 2]
```diff
@@ -91,7 +93,12 @@ func CalculateChunkMetrics(chunk *encoding.Chunk, codecVersion encoding.CodecVer
 		metrics.L1CommitUncompressedBatchBytesSize, metrics.L1CommitBlobSize, err = codecv2.EstimateChunkL1CommitBatchSizeAndBlobSize(chunk)
 		metrics.EstimateBlobSizeTime = time.Since(start)
 		if err != nil {
-			return nil, fmt.Errorf("failed to estimate codecv2 chunk L1 commit blob size: %w", err)
+			if errors.Is(err, &encoding.CompressedDataCompatibilityError{}) {
+				log.Error("Compressed data compatibility check failed", "error", err)
```
Comment (Thegaram) [inline rollup/internal/utils/utils.go]: Why log an error here?

First of all this is already logger in `da-codec`.

Second, we don't know here if this is an error. We'll inspect and log accordingly in higher layers (chunk-proposer, etc.)

**Review by Thegaram — COMMENTED** [review]
LGTM, let's update deps once the other PR is merged.

File: rollup/internal/utils/utils.go (hunk 2 above)
Comment (colinlyguo) [reply rollup/internal/utils/utils.go]: remove the `log.Error` and add `log.Warn` logs both in `chunk-proposer` and `batch-proposer` to print more info to track the chunk & batch proposed due to violation.

**Comment by colinlyguo** [comment]
> LGTM, let's update deps once the other PR is merged.

updated.

**Comment by colinlyguo** [comment]
some issues in unit tests, debugging.

**Review by georgehao — APPROVED** [review]

**Review by Thegaram — APPROVED** [review]

**Pull request merged**